test/spinbox-test.js

```diff
@@ -285,6 +285,21 @@ define( function ( require ) {
 
 	} );
 
+	QUnit.test('spinbox should trigger change after using setValue', function (assert) {
+		var $spinbox = $(html).find('#MySpinboxDecimal').spinbox({
+			value: '1'
+		});
+
+		$spinbox.spinbox('setValue', '2');
+
+		$spinbox.on('changed.fu.spinbox', function () {
+			assert.ok(true, 'spinbox triggers changed after input' );
```

Comment:
This is probably not even getting validated.  You need to use `assert.async` in here and call it to ensure this assertion gets asserted. This is an async test, so you need to tell QUnit it is async.